app/models/post.rb

```diff
@@ -57,6 +57,34 @@ def address
     [street, city, state, postal_code, country].compact.join(", ")
   end
 
+  def self.filter_type(type)
+    if type == "needed" || type == "available"
+      send(type.to_sym)
```

Comment:
Just making our conditional shorter, and DRYing it up. I should make this less clever though.